### ISSUE TITLE
test: add Horn dim-rejection tests for PyTorch, JAX, and MLX

### ISSUE THREAD

**Issue by hunter-heidenreich** [hunter-heidenreich/Kabsch-Cookbook#163]
## Problem

All five frameworks raise `ValueError` when Horn/Horn-with-scale receive non-3D inputs (`dim != 3`). However, this validation is only tested for:

- **NumPy**: `TestNumpySinglePointRejection` in `test_error_handling.py` (tests N=1, not dim!=3)
- **TensorFlow eager**: `test_horn_dim_not_3_rejected_eager` in `test_tf_dynamic_validation.py`

PyTorch, JAX, and MLX Horn dim-rejection is **untested**. The adapter-based `test_raises_error_when_dims_differ` in `test_error_handling.py` tests D(P) != D(Q) mismatch, but not the Horn-specific "D must be 3" check.

## Proposal

Add a test to `test_error_handling.py`:

```python
@pytest.mark.parametrize("algo", ["horn", "horn_with_scale"])
@pytest.mark.parametrize("adapter", frameworks)
def test_horn_raises_error_for_non_3d_input(self, algo, adapter):
    """Horn algorithms must reject inputs where D != 3."""
    P = adapter.convert_in(np.random.default_rng(42).standard_normal((5, 4)).astype("float32"))
    Q = adapter.convert_in(np.random.default_rng(42).standard_normal((5, 4)).astype("float32"))
    func = adapter.get_transform_func(algo)
    with pytest.raises(ValueError, match="3"):
        func(P, Q)
```

This should also test `dim=2` to verify rejection at both sides of the boundary.

## Success criteria

- [ ] `horn(P_4d, Q_4d)` raises `ValueError` tested for all 4 autodiff frameworks
- [ ] `horn(P_2d, Q_2d)` raises `ValueError` tested for all 4 autodiff frameworks
- [ ] Same for `horn_with_scale`
- [ ] Tests use the adapter pattern (not direct imports) for consistency
